### PR TITLE
Move rejection sampling to its own directory

### PR DESCRIPTION
**Pull request by nouhadziri**
This is a simple PR to move the rejection sampling code to its own directory, separate from open-instruct. I can presume the codebase will evolve with additional filtering, generation, and evaluation features, so it is more efficient to keep this code in a dedicated directory. It will eventually make the codebase easier to navigate and maintain.

 Thoughts @vwxyzjn and @natolambert ?

### CODE REVIEW

**Review by vwxyzjn — APPROVED** [review]
@nouhadziri I think this makes sense, especially rejection sampling scripts are all related together. Feel free to merge main and then merge.

Note that you may want to 1) change the docs to line up the correct directory and 2) modify `scripts/rejection_sampling_hello_world.bash` and `scripts/rejection_sampling_tulu.bash` to use the changed folder.

**Review by natolambert — APPROVED** [review]
seems fine, yeah

**Comment by nouhadziri** [comment]
Will merge and close this once this PR #252 is approved and merged.

**Pull request closed without merging**
